### ISSUE TITLE
IterableSDK.h:19:9: 'IterableSDK/IterableConstants.h' file not found

### ISSUE THREAD

**Issue by ileitch** [Iterable/swift-sdk#137]
Getting this error when building as a static library. Using e.g `#import IterableConstants.h` works as expected.

Is there something blocking you from converting these remaining bits of ObjC to Swift?

**Comment by tapashmajumder** [comment]
`typealias` defined in Swift is not supported in Objective C. That is why we are defining typedef in Obj C file. This was quite some time back. We will take a look at this again.

**Comment by ZmicierBiesau** [comment]
If we use modular headers instead of frameworks, we will easily get this error.
`target 'swift-sample-app' do
  # Comment the next line if you don't want to use dynamic frameworks
 use_modular_headers! 

  pod 'Iterable-iOS-SDK'

end`

Will definitely lead us to problems.

**Comment by tapashmajumder** [comment]
@ileitch We have removed the file IterableConststants.h and moved them to Swift in the branch linked below. Can you please check if this fixes your problem?

[https://github.com/Iterable/swift-sdk/tree/feature/mob-680-cocoapods](url)

You can try this by setting this in your Podfile
`pod 'Iterable-iOS-SDK', :git => 'https://github.com/Iterable/swift-sdk.git', :branch => 'feature/mob-680-cocoapods'`

@ZmicierBiesau You can also give this branch a try?

**Comment by tapashmajumder** [comment]
@ileitch @ZmicierBiesau  By the way this branch listed is just meant for testing please don't use this in production. This is just to test whether this fix works. You may have to fix some compilation problems since this branch contains unreleased code.

**Comment by ZmicierBiesau** [comment]
@tapashmajumder I've tried your solution, now "missing" error disappeared.
But I'm receiving a new one:
`
Module map file '/Users/user/Library/Developer/Xcode/DerivedData/swift-sample-app-cyslngeklqgnpscugtkxvgregudy/Build/Products/Debug-iphonesimulator/Iterable-iOS-SDK/IterableSDK.modulemap' not found

`

**Comment by tapashmajumder** [comment]
@ZmicierBiesau Thanks for the update. I think the issue you are seeing may be different than what @ileitch is mentioning. Would you be able to create a small sample project to repro the error you are seeing?

**Comment by ZmicierBiesau** [comment]
@tapashmajumder yes, this is what I'm doing. I using just an empty project with podfile above.

**Comment by tapashmajumder** [comment]
@ZmicierBiesau I have just created an empty project and it is working with use_modular_headers!

Which version of Xcode are you using?

**Comment by ZmicierBiesau** [comment]
@tapashmajumder I just recreated an app and now everything is working. Thank you very much.

**Comment by ileitch** [comment]
Can confirm that the change now builds successfully. 

However, some other change in that branch is causing a warning:
```
Ignoring duplicated output file: '.../DerivedData/Build/Products/Debug-iphonesimulator/Bowtie.app/Assets.car' in shell script build phase '[CP] Copy Pods Resources'. This warning represents an extremely serious project misconfiguration and will likely cause some shell scripts in your project to be skipped entirely, leading to other build failures or missing files in the build directory. This will be a hard error in the future. (in target 'Bowtie')
```

It looks like the following resources are now being included?

```
install_resource "${PODS_ROOT}/Iterable-iOS-SDK/swift-sdk/Resources/IterableInboxCell.xib"
install_resource "${PODS_ROOT}/Iterable-iOS-SDK/swift-sdk/Resources/Assets.xcassets"
```

**Comment by tapashmajumder** [comment]
@ileitch Are you able to proceed with the warning or is it causing any other problem down the line?

I believe when you are building a static library vis `use_modular_headers!` resources are included by Cocoapods.

Can you please have your Iterable customer service rep open a Jira ticket with relevant details?

**Comment by ileitch** [comment]
The build completes and the app ran, but I didn't investigate further. I'm currently using this fork as a static library and those resources are not being included.

**Comment by tapashmajumder** [comment]
@ileitch Thanks for the update. Please feel free to open a Jira ticket so that we can take a look at the issues you are having.

**Comment by tapashmajumder** [comment]
Just released version 6.1.1. This removes all Objective C files in our SDK. If this does not solve your issues please have your Iterable customer service rep open a Jira ticket. Thanks!
